packages/driver/src/cypress/command_queue.ts

```diff
@@ -60,336 +60,322 @@ const commandRunningFailed = (Cypress, state, err) => {
   })
 }
 
-export default {
-  create: (state, timeout, whenStable, cleanup, fail, isCy) => {
-    const queue = $queue.create()
-
-    const { get, slice, at, reset, clear, stop } = queue
-
-    const logs = (filter) => {
-      let logs = _.flatten(_.invokeMap(queue.get(), 'get', 'logs'))
+export class CommandQueue extends Queue<Command> {
+  state: any
+  timeout: any
+  whenStable: any
```

Comment:
Yes, it's a function that returns a `Promise` [(code here)](https://github.com/cypress-io/cypress/blob/2c9cb68450ab0f1fc072e5cf6e21f517ba555039/packages/driver/src/cy/stability.ts#L26-L45).

As I said above, it's left out for the part 3. 

I decided to do this later because this PR is big enough for now(+1000/-1000) and there are some hard things like `state` above and other problems hidden by `//@ts-nocheck`.